resources/fetchers/ecr_fetcher.go

```diff
@@ -24,28 +33,69 @@ type ECRResource struct {
 	EcrRepositories
 }
 
-func NewECRFetcher(awsCfg aws.Config, cfg ECRFetcherConfig) (Fetcher, error) {
-	ecr := NewEcrProvider(awsCfg)
+func NewECRFetcher(awsCfg AwsFetcherConfig, cfg ECRFetcherConfig) (Fetcher, error) {
+	ecrProvider := NewEcrProvider(awsCfg.Config)
+
+	privateRepoRegex := fmt.Sprintf(PrivateRepoRegexTemplate, *awsCfg.AccountID, awsCfg.Config.Region)
```

Comment:
I am a bit confused, it seems like that's the only place you are using `AccountID`.
If that's the case, I think you should call `NewAWSIdentityProvider` right here (instead of the code block you pasted on a comment)

Reply:
That's an option.

My idea was that all of the fetchers will receive as parameter `AwsFetcherConfig` our "Cloudbeat" AWS model, which has also the `AccountID`.

More fetchers in the future will need this data.

But I don't mind changing it if I haven't convinced you.

Reply:
We can keep `AwsFetcherConfig` as the wrapper just AccountID should not be a part of this struct.